### PR TITLE
Add Cypress, some tests, and return TourResource from `store`

### PR DESCRIPTION
**Pull request by jxjj**
## Add Cypress
This adds some testing to Camino to help catch regressions and unexpected behavior. Cypress 9 is used rather than Cypress 10, since:
1. Cypress 10, though it's pretty, seems to need a lot more clicks and is a bit slow.
2. the laracasts/cypress package doesn't yet support 10.

## Some Tests
Also, this PR adds some light test coverage for:
- Home Page
- MyTours page
- creating a tour
- stub tests for other pages.

## Fixes for issues caught by tests
Finally, in the process of testing, two issues were fixed:
- Previously, the initial location for a newly created tour was `null` which caused some issues. Now, a default location (UMN campus) is set when a tour is created.
- The `store` method of `EditTourController` was returning the tour object, which differs slightly from the `TourResource` returned by the API. Specifically, a `Tour` model has a Point object of form:
```json
    "start_location": {
        "type": "Point",
        "coordinates": [
            44.975876,
            -93.234375
        ]
    }
```

Rather than a LngLat  point like the API returns:
```json
 "start_location": {
    "lat": -93.234375,
    "lng": 44.975876
  },
```

This PR transforms the Tour into a TourResource, akin to the API response. It also renames the tour resource class as `TourResource` for clarity.

### CODE REVIEW

File: resources/views/base.blade.php [hunk 1]
```diff
@@ -168,7 +168,7 @@ class="postal-code">55455</span></div>
   window.pusherKey = '{{ env('PUSHER_APP_KEY') }}'
 </script>
 <script>
-  window.mapbox = "{{ config('services.mapbox') }}";
+  window.mapbox = "{{ env('MIX_MAPBOX_ACCESS_TOKEN') }}";
```
Comment (cmcfadden) [inline resources/views/base.blade.php:171]: what's the reason for this change? (Just curious, nbd either way)

**Review by cmcfadden — APPROVED** [review]

File: resources/views/base.blade.php (hunk 1 above)
Comment (jxjj) [reply resources/views/base.blade.php:171]: I'll revert – it's not needed in this PR.

This was changed in the course of troubleshooting why tiles weren't rendering for mapbox sometimes. Turns out we have two env variables for mapbox and I was missing one. `config('services.mapbox')` added another layer, so it got simplified while debugging/

Ultimately, I'd like to just have one env variable, unless a separate one is needed for server side, but that's for later.
